### PR TITLE
fix insert into table select data lose

### PR DESCRIPTION
**Pull request by nicelulu**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):

- Bug Fix

Short description (up to few sentences):

If SquashingBlockOutputStream enable flush, will likely lead to read the data cannot be written to.
Don't interrupt copyData() used inside INSERT SELECT.

### CODE REVIEW

**Comment by nicelulu** [comment]
https://github.com/yandex/ClickHouse/issues/6834

File: dbms/tests/queries/0_stateless/01009_insert_select.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Set any_join_distinct_right_table_keys=1;
+DROP TABLE IF EXISTS test.test_insert_t1;
```
Comment (abyss7) [inline dbms/tests/queries/0_stateless/01009_insert_select.sql]: Do not use `test.` database explicitly in tests.

**Review by abyss7 — CHANGES_REQUESTED** [review]
Remove `test.` prefix in tests

**Comment by abyss7** [comment]
Can't figure out what caused the problem in the first place. Can you explain it a little bit more?

File: dbms/tests/queries/0_stateless/01009_insert_select.sql (hunk 1 above)
Comment (nicelulu) [reply dbms/tests/queries/0_stateless/01009_insert_select.sql]: I'll remove it.

**Comment by nicelulu** [comment]
> Can't figure out what caused the problem in the first place. Can you explain it a little bit more?

Perform an INSERT SELECT. When there is an asynchronous read data like UnionBlockInputStream in the execution plan, the data production is slow in some cases, and nothing can be read in the copyData. This situation will trigger flush, SquashingBlockOutputStream will not write any data after flushing.

**Comment by KochetovNicolai** [comment]
I think I have fixed this issue in another way. @nicelulu can you check if #6911 fixes your case?

**Comment by alexey-milovidov** [comment]
@KochetovNicolai Could you please also import the test from this PR?

**Comment by nicelulu** [comment]
> I think I have fixed this issue in another way. @nicelulu can you check if #6911 fixes your case?

**Pull request closed without merging**